### DISCUSSION TITLE
Mysql2::Error: Data too long for column 'output' at row 1

### DISCUSSION THREAD

**Discussion by haarhoff-frs** [postalserver/postal#2542] [Help with using Postal] [answered]
## Describe the bug

Postal is not able so send some of the messages originating from our ERP system. The error in the backend is:
```
Mysql2::Error: Data too long for column 'output' at row 1

Support Ref: J-HL7V7EH4CN
```
We are using version 2.1.4, and we are not using webhooks. The emails are usually just short text-only messages.
Unfortunately, this was unnoticed for quite a while, and a lot of addresses ended up on the suppression list. Is there any way to clear the suppression list, after the issue is found?

## To Reproduce

Unfortunately it is not quite clear what triggers this. Only emails sent from our ERP have this problem.

## Expected behaviour

I expect them to be sent normally, as all the mails from our webshops etc. do.

## Screenshots
![fmcrederror](https://github.com/postalserver/postal/assets/89089047/83f48e74-9381-47a9-bdda-2da10ccb1cb4)

## Environment details 

 - OS: Ubuntu 22.04
 - Browser Edge (Chromium)
 - Version 114.0.1823.67
 - Type Desktop

## Additional information/context

This is an excerpt of the worker log:
```
postal-worker-1  | [9] [2023-07-10T08:29:58.706] INFO -- : [GMMZ3O97GA] [3::26745 44247] Internal error: Mysql2::Error: Data too long for column 'output' at row 1
postal-worker-1  | [9] [2023-07-10T08:29:58.706] INFO -- : [GMMZ3O97GA] [3::26745 44247] /usr/local/bundle/gems/mysql2-0.5.5/lib/mysql2/client.rb:151:in `_query'
postal-worker-1  | [9] [2023-07-10T08:29:58.706] INFO -- : [GMMZ3O97GA] [3::26745 44247] /usr/local/bundle/gems/mysql2-0.5.5/lib/mysql2/client.rb:151:in `block in query'
postal-worker-1  | [9] [2023-07-10T08:29:58.706] INFO -- : [GMMZ3O97GA] [3::26745 44247] /usr/local/bundle/gems/mysql2-0.5.5/lib/mysql2/client.rb:150:in `handle_interrupt'
postal-worker-1  | [9] [2023-07-10T08:29:58.706] INFO -- : [GMMZ3O97GA] [3::26745 44247] /usr/local/bundle/gems/mysql2-0.5.5/lib/mysql2/client.rb:150:in `query'
postal-worker-1  | [9] [2023-07-10T08:29:58.706] INFO -- : [GMMZ3O97GA] [3::26745 44247] /opt/postal/app/lib/postal/message_db/database.rb:319:in `query_on_connection'
postal-worker-1  | [9] [2023-07-10T08:29:58.706] INFO -- : [GMMZ3O97GA] [3::26745 44247] /opt/postal/app/lib/postal/message_db/database.rb:223:in `block in insert'
postal-worker-1  | [9] [2023-07-10T08:29:58.706] INFO -- : [GMMZ3O97GA] [3::26745 44247] /opt/postal/app/lib/postal/message_db/mysql.rb:20:in `client'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /opt/postal/app/lib/postal/message_db/database.rb:342:in `with_mysql'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /opt/postal/app/lib/postal/message_db/database.rb:222:in `insert'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /opt/postal/app/lib/postal/message_db/delivery.rb:8:in `create'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /opt/postal/app/lib/postal/message_db/message.rb:124:in `create_delivery'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /opt/postal/app/app/jobs/unqueue_message_job.rb:413:in `block in perform'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /opt/postal/app/app/jobs/unqueue_message_job.rb:31:in `each'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /opt/postal/app/app/jobs/unqueue_message_job.rb:31:in `perform'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /opt/postal/app/lib/postal/worker.rb:64:in `receive_job'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /opt/postal/app/lib/postal/worker.rb:96:in `block in join_queue'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/consumer.rb:56:in `call'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/channel.rb:1842:in `block in handle_frameset'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/consumer_work_pool.rb:108:in `block (2 levels) in run_loop'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/consumer_work_pool.rb:103:in `loop'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/consumer_work_pool.rb:103:in `block in run_loop'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/consumer_work_pool.rb:102:in `catch'
postal-worker-1  | [9] [2023-07-10T08:29:58.707] INFO -- : [GMMZ3O97GA] [3::26745 44247] /usr/local/bundle/gems/bunny-2.20.3/lib/bunny/consumer_work_pool.rb:102:in `run_loop'
postal-worker-1  | [9] [2023-07-10T08:29:58.711] INFO -- : [GMMZ3O97GA] [3::26745 44247] Queued message was unlocked
postal-worker-1  | [9] [2023-07-10T08:29:58.800] INFO -- : [GMMZ3O97GA] Finished processing UnqueueMessageJob job in 4.215977953s
```

**Comment by willpower232** [comment]
Do you know where the messages were being sent to? Is it possible that there is some error being output that you can't see by that service?

The field is limited to 512 characters so I imagine there is some kind of issue being reported that you can't see or they are making a large response to you for some reason.

Technically there could be an issue with one of the mentions of output here that needs truncating

https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/lib/postal/smtp_sender.rb#L121

like this

https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/lib/postal/message_db/delivery.rb#L51

**Reply by haarhoff-frs** [reply]
All instances I've seen were sent to customers of us. Nearly all of them have an .gr or .mk TLD on their address, though I'm not sure why that would matter. It's just something I noticed.
Edit: misunderstood
So I'll try to truncate the content that gets set to output in every instance in the send_message method. I'll be back with the results.

**Comment by haarhoff-frs** [comment]
Okay, there seems to be a problem with the receiving mail server:

No SMTP servers were available for ote.gr. Tried mail6.dt-one.com, mail5.dt-one.com, mail8.dt-one.com, mail3.dt-one.com, mail7.dt-one.com, mail2.dt-one.com, mail1.dt-one.com, and mail4.dt-one.com
```
SSL_connect returned=1 errno=0 peeraddr=195.167.xx.xx1:25 state=error: dh key too small, SSL_connect returned=1 errno=0 peeraddr=195.167.xx.xx2:25 state=error: dh key too small, SSL_connect returned=1 errno=0 peeraddr=195.167.xx.xx3:25 state=error: dh key too small, SSL_connect returned=1 errno=0 peeraddr=195.167.xx.xx4:25 state=error: dh key too small, SSL_connect returned=1 errno=0
```
This seems to be a greek internet provider (https://www.ote.gr/), but even on their regular website the SSL certificate has expired. 
Sooo .. not much to do about besides always truncating the error message I guess :( 

Apart from that, is there any way to get an address off the supression list?

**Reply by haarhoff-frs** [reply]
I have truncated the error response truncated by postal, I imagine there are quite a few additional mail servers with invalid certificates that postal went through ...

**Reply by willpower232** [reply, marked as answer]
> dh key too small

We've seen that before, have a look at https://github.com/postalserver/postal/discussions/1558

The only ways to get an email address off the suppression list are either to wait 30 days or have a successful email message sent to them.

I suggest you make the change proposed in that discussion and try releasing an email to the affected email addresses.

**Reply by haarhoff-frs** [reply]
Thanks, that worked. I've added all the changes (the truncation and the openssl settings) into a fork and published my own image, to be on the safe side. For now, everything is working smoothly. Thanks a lot for your support!
